### PR TITLE
apply constructors, seperate elements from bindings

### PR DESCRIPTION
**Pull request by bennguyennz**

### CODE REVIEW

File: Onboarding/Onboarding/Pages/ProfilePages/Contact.cs [hunk 1]
```diff
@@ -13,15 +13,60 @@ namespace Onboarding.Pages.ProfilePages
...
+        Contact ContactObj;
+        public Contact(IWebDriver _driver)
         {
-            string messageXpath = "//div[@class='ns-box-inner']";
-            WaitHelpers.WaitToBeVisible(driver, "XPath", messageXpath, 3);
-            IWebElement message = driver.FindElement(By.XPath(messageXpath));
+          this.driver = _driver;
+        }
+
+        public void ContactStepDefinitions()
```
Comment (amanbansalnz) [inline Onboarding/Onboarding/Pages/ProfilePages/Contact.cs]: This I believe is not needed is there any reason you are doing this?

File: Onboarding/Onboarding/Pages/ProfilePages/Certification.cs [hunk 2]
```diff
@@ -12,187 +12,183 @@ namespace Onboarding.Pages.ProfilePages
...
         {
-            //Wait
-            string xpValue = "//div[@data-tab='fourth']//div[@class ='ui teal button ']";
-            WaitHelpers.WaitToBeClickable(driver, "XPath", xpValue, 5);
+            //initial driver object
+            this.driver = _driver;   
+        }
+
+        //Inintial object and link to constructor 1 by calling method
+        public void CertificationStepDefinitions()
```
Comment (amanbansalnz) [inline Onboarding/Onboarding/Pages/ProfilePages/Certification.cs]: This I believe is not needed is there any reason you are doing this?

File: Onboarding/Onboarding/Pages/ProfilePages/Education.cs [hunk 3]
```diff
@@ -12,141 +12,167 @@ namespace Onboarding.Pages.ProfilePages
...
     {
-        public void AddEducation(IWebDriver driver, string country, string university, string title, string degree, string graduationYear)
+        public IWebDriver driver;
+        Education EducationObj;
+        public Education(IWebDriver _driver)
+        {
+            this.driver = _driver;
+        }
+
+        public void EducationStepDefinitions()
```
Comment (amanbansalnz) [inline Onboarding/Onboarding/Pages/ProfilePages/Education.cs]: This I believe is not needed is there any reason you are doing this?

File: Onboarding/Onboarding/Pages/ProfilePages/Language.cs [hunk 4]
```diff
@@ -12,121 +12,125 @@ namespace Onboarding.Pages.ProfilePages
...
     public class Language
     {
-        public void AddALanguage(IWebDriver driver, string language, string languagelevel)
+        public IWebDriver driver;
+        Language LanguageObj;
+        public Language(IWebDriver _driver)
+        { 
+            this.driver = _driver;
+        }
+        public void LanguageStepDefinitions()
```
Comment (amanbansalnz) [inline Onboarding/Onboarding/Pages/ProfilePages/Language.cs]: This I believe is not needed is there any reason you are doing this?

File: Onboarding/Onboarding/Pages/ProfilePages/Other.cs [hunk 5]
```diff
@@ -12,13 +12,33 @@
...
-        public string getWecomeText(IWebDriver driver)
+        public IWebDriver driver;
+        Other OtherObj;
+
+        public Other(IWebDriver _driver)
+        {
+            this.driver = _driver;
+            
+        }
+        public void OtherStepDefinitions()
```
Comment (amanbansalnz) [inline Onboarding/Onboarding/Pages/ProfilePages/Other.cs]: This I believe is not needed is there any reason you are doing this?

File: Onboarding/Onboarding/Pages/ProfilePages/Skill.cs [hunk 6]
```diff
@@ -12,127 +12,133 @@ namespace Onboarding.Pages.ProfilePages
...
+        public IWebDriver driver;
+        Skill SkillObj;
+        public Skill(IWebDriver _driver)
         {
-            string messageXpath = "//div[@class='ns-box-inner']";
-            WaitHelpers.WaitToBeVisible(driver, "XPath", messageXpath, 3);
-            IWebElement message = driver.FindElement(By.XPath(messageXpath));
+            this.driver = _driver;
+        }
+        public void SkillsStepDefinition()
```
Comment (amanbansalnz) [inline Onboarding/Onboarding/Pages/ProfilePages/Skill.cs]: This I believe is not needed is there any reason you are doing this?

File: Onboarding/Onboarding/StepDefinitions/CertificationStepDefinitions.cs [hunk 7]
```diff
@@ -11,105 +12,109 @@ namespace Onboarding.StepDefinitions
...
+        Certification CertificationObj;
+        LoginPage LoginPageObj;
 
-        Certification CertificationObj = new Certification();
         [Given(@"I signed into the portal")]
         public void GivenISignedIntoThePortal()
         {
             //Open Chrome browser
             driver = new ChromeDriver();
+            CertificationObj = new Certification(driver);
```
Comment (amanbansalnz) [inline Onboarding/Onboarding/StepDefinitions/CertificationStepDefinitions.cs:30]: here you can create a constructor  like so

public CertificationStepDefinitions{
            CertificationObj = new Certification(driver);

}

File: Onboarding/Onboarding/StepDefinitions/CertificationStepDefinitions.cs [hunk 8]
```diff
@@ -11,105 +12,109 @@ namespace Onboarding.StepDefinitions
...
+        //Define Pages and Objects
+        Certification CertificationObj;
+        LoginPage LoginPageObj;
 
-        Certification CertificationObj = new Certification();
         [Given(@"I signed into the portal")]
         public void GivenISignedIntoThePortal()
         {
             //Open Chrome browser
             driver = new ChromeDriver();
```
Comment (amanbansalnz) [inline Onboarding/Onboarding/StepDefinitions/CertificationStepDefinitions.cs:29]: also drive you can move to the Common class and initialise it once and use it 


namespace MarsQA.Utilities
{
    public class CommonDriver
    {
        public static IWebDriver driver;
        LoginPage loginpageObj = new LoginPage();

        [OneTimeSetUp]
        public void onTimeSetup()
        {
            // Open chrome browser
            driver = new ChromeDriver();

            // Login page object initialization and definition you dont have to do this here if you don't want too
            loginpageObj.LoginSteps(driver); // this part you can do in pages if you want or leave it here 

        }


        [OneTimeTearDown]
        public void CloseTestRun()
        {
            // Open chrome browser
            driver = new ChromeDriver();

            driver.Quit();
        }
    }
}

Comment (amanbansalnz) [reply Onboarding/Onboarding/StepDefinitions/CertificationStepDefinitions.cs:29]: this way you dont have to do new driver every where as the in the common class diver variable is static
public static IWebDriver driver;

File: Onboarding/Onboarding/StepDefinitions/ContactStepDefinitions.cs [hunk 9]
```diff
@@ -11,73 +11,77 @@ namespace Onboarding.StepDefinitions
...
-        Contact ContactObj = new Contact();
+        Contact ContactObj;
+        LoginPage LoginPageObj;
+        
 
         [Given(@"I logged into the web portal successfully")]
         public void GivenILoggedIntoTheWebPortalSuccessfully()
         {
             //Open Chrome browser
             driver = new ChromeDriver();
```
Comment (amanbansalnz) [inline Onboarding/Onboarding/StepDefinitions/ContactStepDefinitions.cs:25]: same here

File: Onboarding/Onboarding/StepDefinitions/EducationStepDefinitions.cs [hunk 10]
```diff
@@ -12,123 +12,125 @@ namespace Onboarding.StepDefinitions
...
-
-        Education EducationObj = new Education();
+        Education EducationObj;
+        LoginPage LoginPageObj;
 
         [Given(@"I logged into the portal")]
         public void GivenILoggedIntoThePortal()
         {
             //Open Chrome browser
             driver = new ChromeDriver();
```
Comment (amanbansalnz) [inline Onboarding/Onboarding/StepDefinitions/EducationStepDefinitions.cs:27]: same here

File: Onboarding/Onboarding/StepDefinitions/LanguageStepDefinitions.cs [hunk 11]
```diff
@@ -7,101 +7,103 @@
...
-        Language LanguageObj = new Language();
-
+        Language LanguageObj;
+        LoginPage LoginPageObj;
 
         [Given(@"I signed in the portal")]
         public void GivenISignedInThePortal()
         {
             //Open Chrome browser
             driver = new ChromeDriver();
```
Comment (amanbansalnz) [inline Onboarding/Onboarding/StepDefinitions/LanguageStepDefinitions.cs:28]: same here

File: Onboarding/Onboarding/StepDefinitions/OtherStepDefinitions.cs [hunk 12]
```diff
@@ -11,34 +11,35 @@ namespace Onboarding.StepDefinitions
...
-
-        Other OtherObj = new Other();
+        Other OtherObj;
+        LoginPage LoginPageObj;
 
         [Given(@"I signed into the portal successfully")]
         public void GivenISignedIntoThePortalSuccessfully()
         {
             //Open Chrome browser
             driver = new ChromeDriver();
```
Comment (amanbansalnz) [inline Onboarding/Onboarding/StepDefinitions/OtherStepDefinitions.cs:27]: same here

File: Onboarding/Onboarding/Utilities/CommonDriver.cs [hunk 13]
```diff
@@ -14,14 +14,16 @@ namespace Onboarding.Utilities
     public class CommonDriver
     {
         public IWebDriver driver;
```
Comment (amanbansalnz) [inline Onboarding/Onboarding/Utilities/CommonDriver.cs:16]: so you have done what I said you, I think you don't have to change that to static it should work, try it out!

File: Onboarding/Onboarding/Pages/ProfilePages/Certification.cs (hunk 2 above)
Comment (bennguyennz) [reply Onboarding/Onboarding/Pages/ProfilePages/Certification.cs]: Thanks. It's redundant.

File: Onboarding/Onboarding/Pages/ProfilePages/Contact.cs (hunk 1 above)
Comment (bennguyennz) [reply Onboarding/Onboarding/Pages/ProfilePages/Contact.cs]: `redundant

File: Onboarding/Onboarding/Pages/ProfilePages/Education.cs (hunk 3 above)
Comment (bennguyennz) [reply Onboarding/Onboarding/Pages/ProfilePages/Education.cs]: redundant

File: Onboarding/Onboarding/Pages/ProfilePages/Language.cs (hunk 4 above)
Comment (bennguyennz) [reply Onboarding/Onboarding/Pages/ProfilePages/Language.cs]: redundant

File: Onboarding/Onboarding/Pages/ProfilePages/Other.cs (hunk 5 above)
Comment (bennguyennz) [reply Onboarding/Onboarding/Pages/ProfilePages/Other.cs]: redundant

File: Onboarding/Onboarding/Pages/ProfilePages/Skill.cs (hunk 6 above)
Comment (bennguyennz) [reply Onboarding/Onboarding/Pages/ProfilePages/Skill.cs]: redundant

File: Onboarding/Onboarding/StepDefinitions/CertificationStepDefinitions.cs (hunk 8 above)
Comment (bennguyennz) [reply Onboarding/Onboarding/StepDefinitions/CertificationStepDefinitions.cs:29]: done as no 'static'

File: Onboarding/Onboarding/StepDefinitions/CertificationStepDefinitions.cs (hunk 7 above)
Comment (bennguyennz) [reply Onboarding/Onboarding/StepDefinitions/CertificationStepDefinitions.cs:30]: done

File: Onboarding/Onboarding/StepDefinitions/ContactStepDefinitions.cs (hunk 9 above)
Comment (bennguyennz) [reply Onboarding/Onboarding/StepDefinitions/ContactStepDefinitions.cs:25]: done

File: Onboarding/Onboarding/StepDefinitions/EducationStepDefinitions.cs (hunk 10 above)
Comment (bennguyennz) [reply Onboarding/Onboarding/StepDefinitions/EducationStepDefinitions.cs:27]: done

File: Onboarding/Onboarding/StepDefinitions/LanguageStepDefinitions.cs (hunk 11 above)
Comment (bennguyennz) [reply Onboarding/Onboarding/StepDefinitions/LanguageStepDefinitions.cs:28]: done

File: Onboarding/Onboarding/StepDefinitions/OtherStepDefinitions.cs (hunk 12 above)
Comment (bennguyennz) [reply Onboarding/Onboarding/StepDefinitions/OtherStepDefinitions.cs:27]: done

File: Onboarding/Onboarding/Utilities/CommonDriver.cs (hunk 13 above)
Comment (bennguyennz) [reply Onboarding/Onboarding/Utilities/CommonDriver.cs:16]: Done with no static in driver

**Review by amanbansalnz — APPROVED** [review]

**Pull request merged**